modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIMANAGER4081PaginationCountTestCase.java

```diff
@@ -0,0 +1,219 @@
+            int countInSecondPage = getPaginationElementsCount(storeURLHttp, loginResponseCookie, 10, offset);
+            //fetch the 3nd page, this page should have 4 results
+            int countInThirdPage = getPaginationElementsCount(storeURLHttp, loginResponseCookie, 20, offset);
+
+            //testing the output; added 24 APIs should be paginated as 10,10,4
+            if (countInFirstPage == 10 && countInSecondPage == 10 && countInThirdPage == 4) {
+                isPaginationCorrect = true;
+            }
+        }
+        assertTrue(isPaginationCorrect, "Incorrect Pagination during API Search. Pagination count test case failed.");
+    }
+
+    /**
+     * This method is used to get the number of APIs to be displayed in each page
+     *
+     * @param storeUrl url of the API store
+     * @param loginResponseCookie login response
+     * @param start starting index of the API list
+     * @param offset offset from the starting index of the API list
+     * @return number of APIs in each page
+     */
+    private int getPaginationElementsCount(String storeUrl, String loginResponseCookie, int start, int offset){
+
+        String paginationUrl = storeUrl + "store/site/blocks/api/listing/ajax/list.jag?" +
+                "action=getAllPaginatedPublishedAPIs&" +
+                "tenant=" + tenantDomain +
+                "&start=" + Integer.toString(start) +
+                "&end=" + Integer.toString(offset);
+        int numberOfAPIsInCurrentPage = 0;
+        Map<String, String> paginationRequestHeaders = new HashMap<String, String>();
+        paginationRequestHeaders.put("Cookie", loginResponseCookie);
+        HttpResponse paginationFetchResponse = null;
+
+        try {
+            paginationFetchResponse = HttpRequestUtil.doGet(paginationUrl, paginationRequestHeaders);
+        } catch (IOException e) {
+            //if an error occurs while getting the response, the test is failed
+            assertTrue(false, "Retrieving response for Pagination count test case has failed.");
+        }
+
+        String dataText = "";
+        if(paginationFetchResponse != null) {
+            dataText = paginationFetchResponse.getData();
+        }
+
+        JsonElement paginationDataElement = new JsonParser().parse(dataText);
+        JsonObject paginationDataObject = paginationDataElement.getAsJsonObject();
+
+        // the data in response is evaluated further only if the response indicates successful API retrieval
+        // that is response for the "error" tag should be "false"
+        if ("false".equals(paginationDataObject.get("error").toString())) {
+            numberOfAPIsInCurrentPage = paginationDataObject.get("apis").getAsJsonArray().size();
+        }
+        return  numberOfAPIsInCurrentPage;
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void destroy() throws Exception {
+        super.cleanUp(gatewayContext.getContextTenant().getTenantAdmin().getUserName(),
```

Comment:
Please delete all added APIs in test method within cleanup. :+1:. As a best practice, when after test is completed, system should be back to state that it was just before the test started. This is because some other tests may get failed due to remains of this test resources.